tools/springboard_search.sh

```diff
@@ -125,11 +125,22 @@ function read_config()
 	fi
 }
 
+function get_stack_layout_X86_64()
+{
+	echo $schedule_asm | awk '{for(i = 0; i <= NF; i++) if($i == "push") {print $(i+1);break;}}'
+}
+
+function get_stack_layout_AArch64()
+{
+	echo $schedule_asm | awk '{for(i = 0; i <= NF; i++) if($i == "stp") {print $(i+1),$(i+2);break;}}'
+}
+
 function do_search()
 {
 	read_config
 	eval $(get_function_range __schedule)
 	schedule_asm="$(get_function_asm)"
+	get_stack_layout_$arch > $(dirname $object)/kernel/sched/mod/.vmlinux_stack_layout
```

Comment:
It's kind of bad to keep it in a temp file. Suppose the user uses git for version control. They push `kernel/sched/mod/` to their repo. Files prefixed with "." are ignored by git in most cases. But in fact it will become an essential file for compiling the module after this PR.

I know we haven't documented of advising on version control yet. But how about let's not making it worse for now?

We've tried using modpath in most places to replace the hardcoded "/kernel/sched/mod". So let's stick to tradition and avoid hardcoding paths as much as possible.

Maybe we get rid of these Makefile parameters, and write them in official header files, instead of in Makefile or some temporary file, for example,

context_switch_vmlinux.h:
```c
#define STACK_SIZE_VMLINUX xxx
#define SPRINGBOARD_VMLINUX xxx
#define STACK_LAYOUT_VMLINUX xxx
```

context_switch_module.h:
```c
#define STACK_SIZE_MODULE xxx
#define SPRINGBOARD_MODULE xxx
#define STACK_LAYOUT_MODULE xxx
```

And in main.c, we do something like
```c
COMPILE_ASSERT(COMPILE_STR_EQUAL(STACK_LAYOUT_VMLINUX, STACK_LAYOUT_MODULE))
```

Reply:
This is a good idea. But I think, it is best not to put these macros in a separate header file. If then, STACK_LAYOUT_MODULE becomes a fixed value. So, getting it dynamically in the makefile is OK.